working/1426-extension-types/feature-specification-views.md

```diff
@@ -74,24 +197,24 @@ reasoning is required may be fragmented into many different locations, and
 there is no help detecting that some of those locations are treating the
 tree incorrectly according to the schema.
 
-If views are supported, we can declare a set of view types with
-operations that are tailored to work correctly with the given schema and
-its subschemas. This is less error-prone and more maintainable than the
-approach where the tree is handled with static type `dynamic` everywhere.
+If views are available then we can declare a set of view types with
+operations that are tailored to work correctly with the given schema
+and its subschemas. This is less error-prone and more maintainable
+than the approach where the tree is handled with static type `dynamic`
+everywhere.
 
 Here's an example that shows the core of that scenario. The schema that
 we're assuming allows for nested `List<dynamic>` with numbers at the
 leaves, and nothing else.
 
 ```dart
-view TinyJson on Object {
+view TinyJson(Object it) {
   Iterable<num> get leaves sync* {
-    var self = this;
-    if (self is num) {
-      yield self;
-    } else if (self is List<dynamic>) {
-      for (var element in self) {
-        yield* element.leaves;
+    if (it is num) {
```

Comment:
just mentioning in passing that there's an implicit assumption of promotability here.  Probably warranted, but requires we specify the special treatment here.